### DISCUSSION TITLE
Tools for development

### DISCUSSION THREAD

**Discussion by depascalis** [minetest-mods/i3#67] [Q&A] [answered]
I was wondering if you could share the tools used during development such as preprocessors etc. so that one could load this repo without drowning in errors and warnings? It would also be helpful to be able to track definitions of imported functions etc.

I'm using vscode on Windows if that makes any difference.

Thanks ♥

**Comment by kilbith** [comment, marked as answer]
Hey,

You need a specific Luacheck setup in order to lint this codebase : https://github.com/minetest-mods/i3/blob/main/util/luacheck.lua

The preprocessor has been designed for my own use since it's a one-man-made project, thus it wasn't meant to be shared and I don't think much would be interested in it.

Good luck !

**Reply by depascalis** [reply]
Oh so it was there all the time, I'm completely new to Lua so I was unsure about that one. Thanks for the help!
